optimizely/optimizely_user_context.py

```diff
@@ -12,17 +12,27 @@
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
 #
-
+from __future__ import annotations
 import copy
 import threading
+from typing import Any, Optional, TYPE_CHECKING
+
+from optimizely.decision import optimizely_decision
+
+if TYPE_CHECKING:
+    # prevent circular dependency
+    from optimizely.optimizely import Optimizely
```

Comment:
Can you explain this if statement? 
Why did you use TYPE_CHECKING constant here? Is it to import optimizely.optimizely only when we run mypy (i.e. when type checking is in progress)?)

Reply:
Yea that's right, `TYPE_CHECKING` is True when type checking and False at runtime. The common use case is to allow you to import a specific class/type instead of just the module, when importing the class/type causes a circular dependency. That way the type hints don't have to be `optimizely.Optimizely`.

Although in this specific file, Optimizely is only used once, so I could see an argument being made that it is unnecessary here.

Reply:
ok, I think it's ok to leave it. It makes its use explicit and transparent